styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -141,24 +142,38 @@ void tick() {
       if (resourceRefs.isEmpty()) {
         sendDequeue(instance);
       } else {
-        final boolean canRun = resourceRefs.stream()
-            .allMatch(resourceId -> {
+        final Set<String> unknownResources = Sets.newHashSet();
+        final Set<Resource> depletedResources = resourceRefs.stream()
+            .flatMap(resourceId -> {
               if (!resources.containsKey(resourceId)) {
-                // todo: send info event, referencing unknown resource
-                return false;
+                unknownResources.add(resourceId);
+                return Stream.empty();
               }
 
               final Resource resource = resources.get(resourceId);
               final long usage = currentResourceUsage.getOrDefault(resourceId, 0L);
-              return usage < resource.concurrency();
-            });
-
-        if (canRun) {
+              if (usage >= resource.concurrency()) {
+                return Stream.of(resource);
+              }
+              return Stream.empty();
```

Comment:
that's exactly correct! but it returns a `Resource` rather than a `String resourceId`. Sure, could do filter and then map...